### PR TITLE
Site: Add instructions for updating PRs based on the discussion in the dev list (Stamatis Zampetakis)

### PR DESCRIPTION
**Pull request by zabetak**
The complete discussion can be found at https://mail-archives.apache.org/mod_mbox/calcite-dev/201808.mbox/%3CCA%2BEpF8tsJBXEawKTaXNsQ1dqcDBymxey1FdRWHuAtwWj1pHKWw%40mail.gmail.com%3E

### CODE REVIEW

File: site/develop/index.md [hunk 1]
```diff
@@ -88,6 +88,14 @@ your branch to the calcite master branch. Update the JIRA case
 to reference your pull request, and a committer will review your
 changes.
 
+The pull request may need to be updated (after its submission) for three main reasons: (i) you identified a problem after the submission of the pull request; (ii) the reviewer requested further changes; (iii) the Travis CI failed and the failure is not caused by your changes.
+
+In order to update the pull request you need to commit the changes in your branch and then push the commit(s) to github. Your are encouraged to use regular (non-rebase) commits on top of previously existing ones.
```
Comment (vlsi) [inline site/develop/index.md]: GitHub

File: site/develop/index.md [hunk 2]
```diff
@@ -88,6 +88,14 @@ your branch to the calcite master branch. Update the JIRA case
...
+
+When pushing the changes to github you should refrain from using the `--force` parameter and its alternatives. You are allowed to force push your changes under certain conditions: (i) the pull request has been submitted less than 10 minutes ago and there is no pending discussion (in the PR and/or in JIRA) concerning it; (ii) a reviewer has explicitly asked you to perform some modifications that require the use of the `--force` option.
```
Comment (vlsi) [inline site/develop/index.md]: GitHub

File: site/develop/index.md [hunk 3]
```diff
@@ -115,4 +123,3 @@ We value all contributions that help to build a vibrant community, not just code
 You can contribute by testing the code, helping verify a release,
 writing documentation or the web site,
 or just by answering questions on the list.
-
```
Comment (vlsi) [inline site/develop/index.md]: Was it intended?

File: site/develop/index.md (hunk 1 above)
Comment (zabetak) [reply site/develop/index.md]: Fixed

File: site/develop/index.md (hunk 2 above)
Comment (zabetak) [reply site/develop/index.md]: Fixed

File: site/develop/index.md (hunk 3 above)
Comment (zabetak) [reply site/develop/index.md]: Does not appear in regular git diff, I don't know why it appears here.

Comment (vlsi) [reply site/develop/index.md]: Well, it does appear in the console `git diff` for me (macos). Original file ended up with two `0x0a` characters and a fixed one has just one.

On the other hand, let's just fix it.

File: site/develop/index.md [hunk 4]
```diff
@@ -83,11 +83,19 @@ If your change had multiple commits, use `git rebase -i master` to
 squash them into a single commit, and to bring your code up to date
 with the latest on the main line.
 
-Then push your commit(s) to github, and create a pull request from
+Then push your commit(s) to GitHub, and create a pull request from
 your branch to the calcite master branch. Update the JIRA case
 to reference your pull request, and a committer will review your
 changes.
 
+The pull request may need to be updated (after its submission) for three main reasons: (i) you identified a problem after the submission of the pull request; (ii) the reviewer requested further changes; (iii) the Travis CI failed and the failure is not caused by your changes.
```
[the remark this answers is not in the document]
Comment (zabetak) [reply site/develop/index.md]: I didn't intend to but it appears much nicer like that :) I reformatted the text accordingly.

File: site/develop/index.md (hunk 3 above)
Comment (zabetak) [reply site/develop/index.md]: It's my editor who ate the line :P I add it back!

File: site/develop/index.md [hunk 5]
```diff
@@ -88,13 +88,26 @@ your branch to the calcite master branch. Update the JIRA case
...
-
-In the special case, that the Travis CI failed and the failure is not caused by your changes create an empty commit (`git commit --allow-empty`) and push it.
+The pull request may need to be updated (after its submission) for three main
+reasons:
+1. you identified a problem after the submission of the pull request;
+2. the reviewer requested further changes;
+3. the Travis CI failed and the failure is not caused by your changes.
+
+In order to update the pull request you need to commit the changes in your
+branch and then push the commit(s) to GitHub. Your are encouraged to use regular
```
Comment (michaelmior) [inline site/develop/index.md]: "Your are" -> "you are"

File: site/develop/index.md [hunk 6]
```diff
@@ -88,13 +88,26 @@ your branch to the calcite master branch. Update the JIRA case
...
-In order to update the pull request you need to commit the changes in your branch and then push the commit(s) to GitHub. Your are encouraged to use regular (non-rebase) commits on top of previously existing ones.
-
-When pushing the changes to GitHub you should refrain from using the `--force` parameter and its alternatives. You are allowed to force push your changes under certain conditions: (i) the pull request has been submitted less than 10 minutes ago and there is no pending discussion (in the PR and/or in JIRA) concerning it; (ii) a reviewer has explicitly asked you to perform some modifications that require the use of the `--force` option.
-
-In the special case, that the Travis CI failed and the failure is not caused by your changes create an empty commit (`git commit --allow-empty`) and push it.
+The pull request may need to be updated (after its submission) for three main
+reasons:
+1. you identified a problem after the submission of the pull request;
+2. the reviewer requested further changes;
+3. the Travis CI failed and the failure is not caused by your changes.
```
Comment (michaelmior) [inline site/develop/index.md]: I would add "build" to Travis CI

File: site/develop/index.md [hunk 7]
```diff
@@ -88,13 +88,26 @@ your branch to the calcite master branch. Update the JIRA case
...
+
+In order to update the pull request you need to commit the changes in your
```
Comment (michaelmior) [inline site/develop/index.md]: I would add a comma between "request" and "you"

File: site/develop/index.md [hunk 8]
```diff
@@ -88,13 +88,26 @@ your branch to the calcite master branch. Update the JIRA case
...
+branch and then push the commit(s) to GitHub. Your are encouraged to use regular
+ (non-rebase) commits on top of previously existing ones.
```
Comment (michaelmior) [inline site/develop/index.md]: I'd suggest changing "non-rebase" to "non-rebased"

File: site/develop/index.md [hunk 9]
```diff
@@ -88,13 +88,26 @@ your branch to the calcite master branch. Update the JIRA case
...
+
+When pushing the changes to GitHub you should refrain from using the `--force`
```
Comment (michaelmior) [inline site/develop/index.md]: comma between "GitHub" and "you"

File: site/develop/index.md [hunk 10]
```diff
@@ -88,13 +88,26 @@ your branch to the calcite master branch. Update the JIRA case
...
+parameter and its alternatives. You are allowed to force push your changes under
```
Comment (michaelmior) [inline site/develop/index.md]: I would change "You are allowed" to "You may choose". Ultimately the author of the PR is "allowed" to do whatever they want, but we're trying to set some guidelines here.

File: site/develop/index.md [hunk 11]
```diff
@@ -88,13 +88,26 @@ your branch to the calcite master branch. Update the JIRA case
...
+ certain conditions:
+ * the pull request has been submitted less than 10 minutes ago and there is no
+ pending discussion (in the PR and/or in JIRA) concerning it;
+ * a reviewer has explicitly asked you to perform some modifications that
+ require the use of the `--force` option.
+
+In the special case, that the Travis CI failed and the failure is not caused by
```
Comment (michaelmior) [inline site/develop/index.md]: add "build" to "Travis CI"

File: site/develop/index.md [hunk 12]
```diff
@@ -123,3 +136,4 @@ We value all contributions that help to build a vibrant community, not just code
 You can contribute by testing the code, helping verify a release,
 writing documentation or the web site,
 or just by answering questions on the list.
+
```
Comment (michaelmior) [inline site/develop/index.md]: Is this blank line necessary?

File: site/develop/index.md (hunk 5 above)
Comment (zabetak) [reply site/develop/index.md]: Fixed

File: site/develop/index.md (hunk 6 above)
Comment (zabetak) [reply site/develop/index.md]: Fixed

File: site/develop/index.md (hunk 7 above)
Comment (zabetak) [reply site/develop/index.md]: Fixed

File: site/develop/index.md (hunk 8 above)
Comment (zabetak) [reply site/develop/index.md]: Fixed

File: site/develop/index.md (hunk 9 above)
Comment (zabetak) [reply site/develop/index.md]: Fixed

File: site/develop/index.md (hunk 10 above)
Comment (zabetak) [reply site/develop/index.md]: True :)

File: site/develop/index.md (hunk 11 above)
Comment (zabetak) [reply site/develop/index.md]: Fixed

File: site/develop/index.md (hunk 12 above)
Comment (zabetak) [reply site/develop/index.md]: I removed by accident before, so I add it back now.

**Comment by michaelmior** [comment]
@vlsi Since you've looked at this as well, any further changes you think should be made before this is merged?

**Comment by vlsi** [comment]
@michaelmior , English is far from being my native language, so I have no strong comments here.

I think it might make sense to add clarification for reviewers/committers (e.g. when to ask or not to ask the rebase, etc).

**Comment by vlsi** [comment]
Just to clarify: I'm +1 re the current PR.

As usual, more things can be added later.

**Pull request closed without merging**